### PR TITLE
Include anyuid usage to run on OpenShift

### PR DESCRIPTION
**Pull request by sarroutbi**
Resolves: #33

### CODE REVIEW

**Comment by sarroutbi** [comment]
@mheese : I will include an openshift global Helm value for now, and, in the future, I can prepare a change for OpenShift autodetection in Helm (I still need some time to investigate how to do it)

**Comment by mheese** [comment]
@sarroutbi great thing that you are getting that started! :+1: 

I think a flag for openshift is totally fine right now. As I said before I don't even know if it _is_ possible at all to do auto-detection.

A few questions on the changes:

- I assume you need the `anyuid` SCC because of the users that are used in the registrar, verifier and init pods?
- if yes, it might make sense to actually create separate dedicated SCCs that narrow it down _exactly_ for the use case of those pods, and that get only bound to the specific service accounts that we are using for these services.
- Why? the reason for that boils down to regulations/compliance in a lot of companies (which is a bit stupid if you ask me) that prohibit the use of these SCCs. However, they are fine if there are dedicated SCCs for the specific use-cases that also must not be used for anything else.
- I'm curious to see that there is no entry for the tenant yet?
- same goes for the agent

**Comment by sarroutbi** [comment]
Hello @mheese . Thanks for your feedback. Let me answer inline

> @sarroutbi great thing that you are getting that started! 👍
> 
> I think a flag for openshift is totally fine right now. As I said before I don't even know if it _is_ possible at all to do auto-detection.
> 
> A few questions on the changes:
> 
> * I assume you need the `anyuid` SCC because of the users that are used in the registrar, verifier and init pods?

Yes. It is the mechanism I found for registrar, verifier and init pods to run correctly

> * if yes, it might make sense to actually create separate dedicated SCCs that narrow it down _exactly_ for the use case of those pods, and that get only bound to the specific service accounts that we are using for these services.

For that, I would have to figure out what is the exact issue each of them has. I will try, but, if possible, I would like to do this in a different PR. Right now, it is helpful for us to have a deployment that runs on OpenShift, so that we can start our first tests once deployment works correctly.

> * Why? the reason for that boils down to regulations/compliance in a lot of companies (which is a bit stupid if you ask me) that prohibit the use of these SCCs. However, they are fine if there are dedicated SCCs for the specific use-cases that also must not be used for anything else.

OK. I was not aware of that. As said before, I would like to have a "first working deployment" for our QE to start working with this scenario

> * I'm curious to see that there is no entry for the tenant yet?

Tenant runs appropriately, without requiring the usage of a specific SCC

> * same goes for the agent

At this particular moment, we are focused in deployment *without agent*, at least, when deploying in OpenShift.

**Comment by mheese** [comment]
@sarroutbi alright, this makes sense. And I can also see that we should unblock you so that you guys can make progress.

Can you file another issue please that we must review SCCs for the OpenShift deployment and link it to this issue? Then I think we can proceed with this, unless @maugustosilva has any objections.

**Comment by sarroutbi** [comment]
> @sarroutbi alright, this makes sense. And I can also see that we should unblock you so that you guys can make progress.
> 
> Can you file another issue please that we must review SCCs for the OpenShift deployment and link it to this issue? Then I think we can proceed with this, unless @maugustosilva has any objections.

Absolutely. I created #36 to track SCCs fine tuning

File: build/helm/keylime/charts/keylime-registrar/values.yaml [hunk 1]
```diff
@@ -20,6 +20,9 @@ serviceAccount:
   # If not set and create is true, a name is generated using the fullname template
   name: ""
 
+roleName: "registrar-role"
+roleBindingName: "registrar-rolebinding"
```
Comment (mheese) [inline build/helm/keylime/charts/keylime-registrar/values.yaml:24]: @sarroutbi sorry, just seeing this now... can these not be derived from release name / release information?

Comment (mheese) [reply build/helm/keylime/charts/keylime-registrar/values.yaml:24]: basically, similarly to how it's done for service accounts for example if you understand what I mean

Comment (sarroutbi) [reply build/helm/keylime/charts/keylime-registrar/values.yaml:24]: Yes, I think I understand. I will update the PR to include a template similar to the one used in service accounts

**Comment by sarroutbi** [comment]
Hello @mheese . After latest patch, this is how Role/RoleBindings are named:
```bash
$ oc get rolebinding -n keylime
NAME                     ROLE                               AGE
hhkl-keylime-registrar   Role/hhkl-keylime-registrar        33m
hhkl-keylime-verifier    Role/hhkl-keylime-verifier         33m
...
$ oc get role -n keylime
NAME                     CREATED AT
hhkl-keylime-registrar   2023-10-31T09:59:27Z
hhkl-keylime-verifier    2023-10-31T09:59:27Z
...
$ oc get sa -n keylime
NAME                     SECRETS   AGE
builder                  1         55m
default                  1         55m
deployer                 1         55m
hhkl-keylime-agent       1         33m
hhkl-keylime-init        1         34m
hhkl-keylime-registrar   1         33m
hhkl-keylime-tenant      1         33m
hhkl-keylime-verifier    1         33m
```
Thanks for your feedback

**Review by mheese — APPROVED** [review]
lgtm :+1:

**Pull request merged**